### PR TITLE
Refactor Code Tables

### PR DESCRIPTION
**Pull request by dhaselhan**
* Remove ObsoleteBaseCodeEntity
* Change all tables to use code directly
* Update API and DTOs to match

### CODE REVIEW

File: alcs-api/src/application/application.controller.spec.ts [hunk 1]
```diff
@@ -408,6 +370,13 @@ describe('ApplicationController', () => {
       card: updatedCard,
     } as Application);
 
+    applicationService.mapToDtos.mockResolvedValue([
+      {
+        ...mockApplicationEntity,
+        card: updatedCard as any,
```
Comment (mhuseinov) [inline alcs-api/src/application/application.controller.spec.ts:376]: why do you need any everywhere?

File: alcs-api/src/code/code.service.spec.ts [hunk 2]
```diff
@@ -144,7 +140,7 @@ describe('CodeService', () => {
     const mockStatus = {
       uuid: '2',
     };
-    mockStatusRepository.findOne.mockResolvedValue(mockStatus as CardStatus);
+    mockStatusRepository.findOne.mockResolvedValue(mockStatus as any);
```
Comment (mhuseinov) [inline alcs-api/src/code/code.service.spec.ts:143]: the same with any

File: alcs-api/src/application/application.controller.spec.ts (hunk 1 above)
Comment (dhaselhan) [reply alcs-api/src/application/application.controller.spec.ts]: Removed

File: alcs-api/src/code/code.service.spec.ts (hunk 2 above)
Comment (dhaselhan) [reply alcs-api/src/code/code.service.spec.ts]: Changed back to as CardStatus and fixed.

**Review by mhuseinov — APPROVED** [review]
🔥

**Pull request merged**